src/asm/lexer.c

```diff
@@ -1102,11 +1102,11 @@ static void discardComment(void)
 	lexerState->disableInterpolation = false;
 }
 
-/* Function to read a line continuation */
+/* Functions to read a line continuation or separator */
 
-static void readLineContinuation(void)
+static bool readLineContinuationOrSeparator(void)
```

Comment:
A line continuation allows a logical "line" to span several lines. This instead allows a single line to contain several logical "lines", so I disagree with calling it a "line continuation".

Similarly, I disagree with the use of the backslash for it, since depending on the context, it would either continue a single statement, or separate multiple ones. This is further problematic because of EQUS:
```asm
EMPTY equs ""

ld a, 42 \ EMPTY
```
This backslash is treated as a statement separator, but is followed by nothing (= an empty statement). This is weird, and conflicts with the idea that EQUSes are "copy-pasted" into the code, where I would expect the backslash to instead be treated as a line continuation.

Reply:
"`readLineContinuationOrSeparator`" reads a line continuation (a backslash with nothing after it, allowing one statement to span across multiple lines) or a line separator (a backslash with something after it, allowing multiple statements on one line). Should it be called something else?

I see `ld a, 42 \ EMPTY` as a problem with `EQUS` expansion, not with the line separator. It existed even before this PR: you'd get the error "Begun line continuation, but encountered character 'E'". As usual, the problem is that the identifier has to be lexed and *then* expanded, unlike macro arguments and interpolations which can be expanded in `peek` before the current lexer mode sees anything.

Reply:
I don't see a better choice of statement separator than `\`. People already use `\`, `/`, or `;` informally. `/` would be ambiguous with division and `;` obviously won't work. Doubled-up `//` or `\\` would also be valid, but `//` looks like a C comment and `\\` looks escape-related. `:` would also look reasonable, except that it's ambiguous with anonymous labels.

Actually, there's another option: the so-far unused `T_PERIOD`. It would act as a line terminator instead of the artificial `T_EOL`. Would `ld a, [hli] . ld h, [hl] . ld l, a` be better than `ld a, [hli] \ ld h, [hl] \ ld l, a`? (I personally prefer the `\` because it stands out more.)

Reply:
I personally don't mind periods.

Reply:
I'm still strongly against `\` due to the bizarre inverted semantics. I agree that `.` doesn't stand out enough, imo.

At this point, I'm starting to question the relevancy of this syntax: it's only used in chats informally to use "in-line" code block syntax, but is it relevant in an actual code file?

Reply:
I would find it useful to copy+paste code from chat to a file (or to rgbds-live) and have it work without reformatting; or to write some code in that way so it can be quickly pasted into chat after testing.

Longer-term, it would also complement #902 to make it easier for users to adapt if we eliminate `EQUS` expansion. Two of pret's most common (only?) needs for it would be fulfilled by user-defined functions (e.g. `ld bc, 27 tiles` → `ld bc, tiles(27)`) and by macros (e.g. `text EQUS "db 0,"` → `DEF text MACRO db 0, \#`).

I see your point about "inverted semantics", although it's at least unambiguous to lex, and might be clear to read too in practice.

If `DEF . EQUS "\n"` were valid, that would actually make `.` work exactly as desired. I think if `\` isn't acceptable, `.` is next-best, despite not standing out: periods have an obvious meaning as ending sentences, and a few languages like Smalltalk also use them as statement separators.

Reply:
Updated this to use `.` instead of `\`.

Reply:
> I would find it useful to copy+paste code from chat to a file (or to rgbds-live) and have it work without reformatting

While the other (being able to paste from RGBDS to a chat) is valid, this one cannot work for people formatting it as `instr1 ; instr2 ; instr3`, so I'm doubtful.

> I see your point about "inverted semantics", although it's at least unambiguous to lex, and might be clear to read too in practice.

From experience talking with newbies about RGBDS' idiosyncratic behavior, it seems to often make more sense to us maintainers because we're becoming used to reasoning in terms of lexing + parsing, but someone without some language theory background wouldn't think that way.
This is why I'm trying to think differently, though I may be misguided in how I'm doing it.